src/test/LibsodiumEncryptor.test.ts

```diff
@@ -0,0 +1,259 @@
+
+    const { encryptedItem, nonce } = legacyEncrypt({
+      item: account,
+      fieldsToEncrypt,
+      dataKey
+    })
+
+    const { decryptedItem } = encryptor.decrypt({
+      encryptedItem,
+      nonce,
+      dataKey,
+      fieldsToDecrypt: fieldsToEncrypt
+    })
+
+    expect(decryptedItem).toEqual(account)
+  })
+
+  it("should encrypt and decrypt an item with an undefined or null field", async () => {
+    const fieldsToEncrypt: (keyof BankAccount)[] = [
+      "accountNumber",
+      "balance",
+      "routingNumber",
+      "notes",
+      "bankName"
+    ]
+
+    const dataKeyProvider = await FixedDataKeyProvider.forLibsodium()
+    const { dataKey } = await dataKeyProvider.generateDataKey()
+
+    const emptyValues = [undefined, null]
+    emptyValues.forEach((emptyValue) => {
+      const item = { ...account, bankName: emptyValue }
+      const { plaintextFields, nonce, encryptedFields } = encryptor.encrypt({
+        item,
+        fieldsToEncrypt,
+        dataKey
+      })
+
+      const { decryptedItem } = encryptor.decrypt({
+        encryptedItem: plaintextFields,
+        nonce,
+        dataKey,
+        encryptedFields
+      })
+
+      expect(decryptedItem).toEqual(item)
+    })
+  })
+
+  it("should encrypt and decrypt binary fields", async () => {
+    const dataKeyProvider = await FixedDataKeyProvider.forLibsodium()
+    const { dataKey } = await dataKeyProvider.generateDataKey()
+
+    const binaryItem = {
+      name: "hello world",
+      binaryData: Buffer.from("hello world", "utf-8")
+    }
+
+    const { plaintextFields, nonce, encryptedFields } = encryptor.encrypt({
+      item: binaryItem,
+      fieldsToEncrypt: ["name", "binaryData"],
+      dataKey
+    })
+
+    const { decryptedItem } = encryptor.decrypt({
+      encryptedItem: plaintextFields,
+      nonce,
+      dataKey,
+      encryptedFields
+    })
+
+    expect(decryptedItem).toEqual(binaryItem)
+  })
+
+  it("should encrypt and decrypt binary fields recursively", async () => {
+    const dataKeyProvider = await FixedDataKeyProvider.forLibsodium()
+    const { dataKey } = await dataKeyProvider.generateDataKey()
+
+    const binaryItem = {
+      name: "hello world",
+      data: {
+        otherData: {
+          binaryData: Buffer.from("hello world", "utf-8")
+        }
+      }
+    }
+
+    const { plaintextFields, nonce, encryptedFields } = encryptor.encrypt({
+      item: binaryItem,
+      fieldsToEncrypt: ["name", "data"],
+      dataKey
+    })
+
+    const { decryptedItem } = encryptor.decrypt({
+      encryptedItem: plaintextFields,
+      nonce,
+      dataKey,
+      encryptedFields
+    })
+
+    expect(decryptedItem).toEqual(binaryItem)
+  })
+})
+
+/** This is a copy/pasted "legacy" version of LibsodiumEncryptor's encrypt function, which had
+ *  a bug that reused nonces across multiple encrypt operations.
+ *
+ *  It is preserved here for testing to ensure we can decrypt items encrypted with this format
+ *  */
```

Comment:
This makes me feel a bit better that this nonce fix "should" be compatible with items that were encrypted prior to the fix.